src/nmqtt.nim

```diff
@@ -351,11 +358,13 @@ proc onPubRec(ctx: MqttCtx, pkt: Pkt) {.async.} =
   assert ctx.workQueue[msgId].qos == 2
   var pkt = newPkt(PubRel, 0b0010)
   pkt.put(msgId)
-  if await ctx.send(pkt):
-    ctx.workQueue.del msgId
+  discard await ctx.send(pkt)
```

Comment:
Futures should never be discarded.
https://nim-lang.org/docs/asyncdispatch.html#discarding-futures

Reply:
You're right

Reply:
I'll make a pr in your repo.